tethys_portal/templates/header.html

```diff
@@ -27,11 +27,21 @@
       <div id="header-navbar-nav" class="collapse navbar-collapse">
         <ul class="navbar-nav ms-auto">
           {% block header_nav_items %}
-            {% url 'app_library' as app_library_url %}
+            {% if single_app_mode %}
+              {% if configured_single_app %}
+                {% url configured_single_app.index_url as header_url %}
+              {% else %} 
+                {% url 'user:profile' as header_url  %} 
+              {% endif %}
+            {% else %}
+              {% url 'app_library' as header_url %}
+            {% endif %}
             {% if ENABLE_OPEN_PORTAL or user.is_authenticated and user.is_active %}
-              <li class="nav-item">
-                <a class="h5 me-3 mb-0 nav-link{% if request.path == app_library_url %} active" aria-current="page{% endif %}" href="{{ app_library_url }}" title="{% if single_app_name %}{{ single_app_name }}{% elif site_globals.apps_library_title %}{{ site_globals.apps_library_title }}{% else %}Apps{% endif %}">{% if single_app_name %}{{ single_app_name }}{% elif site_globals.apps_library_title %}{{ site_globals.apps_library_title }}{% else %}Apps{% endif %}</a>
-              </li>
+              {% if not single_app_mode or single_app_mode and configured_single_app.name %}
```

Comment:
Move this logic to the context processor and use simple context variables like `app_library_name` and `app_library_url`. Set the values in the context processor.